### PR TITLE
Haddocks for internal modules

### PR DESCRIPTION
**Pull request by coot**
Added --haddock-internl flag `cabal haddock` command.
 Also create the OUTPUT_DIR if it does not exist.

### CODE REVIEW

File: scripts/haddocs.sh [hunk 1]
```diff
@@ -49,6 +50,10 @@ elif [ ${REGENERATE} != "false" ]; then
   cabal haddock "${HADDOCK_OPTS[@]}" ${REGENERATE}
 fi
 
+if [[ !( -d ${OUTPUT_DIR} ) ]]; then
+  mkdir ${OUTPUT_DIR}
+fi
+
```
Comment (karknu) [inline scripts/haddocs.sh:56]: Couldn't this be replaced with `mkdir -p ${OUTPUT_DIR}`?

**Comment by coot** [comment]
bors merge

File: scripts/haddocs.sh (hunk 1 above)
Comment (coot) [reply scripts/haddocs.sh:56]: Yes, that's better.

**Comment by coot** [comment]
borse merge

**Comment by coot** [comment]
bors merge

File: scripts/haddocs.sh (hunk 1 above)
Comment (mrBliss) [reply scripts/haddocs.sh:56]: The point of `-p` is that you no longer need to check whether it exists or not.

**Pull request merged**